File: weektwo/repeat/sortingagain/Bubble.java
package repeat.sortingagain;

import java.util.Arrays;

public class Bubble {
    public static void main(String[] args) {
        int arr[] = { 2, 5, 4, 1, 3 };
        bubble(arr);
        System.out.println(Arrays.toString(arr));
    }

    private static void bubble(int[] arr) {
        boolean swaped;
        for (int i = 0; i < arr.length; i++) {
            swaped = false;
            for (int j = 1; j < arr.length - i; j++) {
                if (arr[j] < arr[j - 1]) {
                    swap(arr, j, j - 1);
                    swaped = true;
                }
            }
            if (!swaped) {
                break;
            }
        }
    }

    private static void swap(int[] arr, int one, int two) {
        int temp = arr[one];
        arr[one] = arr[two];
        arr[two] = temp;
    }
}

File: dsaJavascript/revise/bubble.js
function bubbleSort(arr) {
  let isSwapped = false;
  for (let i = 0; i < arr.length; i++) {
    for (let j = 0; j < arr.length - i - 1; j++) {
      if (arr[j] > arr[j + 1]) {
        swap(arr, j, j + 1);
        isSwapped = true;
      }
    }
    if (!isSwapped) break;
  }
}

function swap(arr, i, j) {
  let temp = arr[i];
  arr[i] = arr[j];
  arr[j] = temp;
}

let arr = [1, 292, 21, 29, 3, -2];

bubbleSort(arr);
console.log(arr);